cloudcix_primitives/vpns2s_ns.py

```diff
@@ -1,23 +1,448 @@
+    return:
+        description: |
+            A tuple with a boolean flag stating the build was successful or not and
+            the output or error message.
+        type: tuple
+    """
+    
+    # Define messages
+    messages = {
+        1000: f'Successfully configured Site-to-Site VPN tunnel {namespace}-{vpn_id}',
+        
+        # For enabled PodNet
+        3021: f'Failed to connect to the enabled PodNet for payload write_config: ',
+        3022: f'Failed to run write_config payload on the enabled PodNet: ',
+        3023: f'Failed to connect to the enabled PodNet for payload terminate_vpn: ',
+        3024: f'Failed to run terminate_vpn payload on the enabled PodNet: ',
+        3025: f'Failed to connect to the enabled PodNet for payload load_vpn: ',
+        3026: f'Failed to run load_vpn payload on the enabled PodNet: ',
+        3027: f'Failed to connect to the enabled PodNet for payload initiate_vpn: ',
+        3028: f'Failed to run initiate_vpn payload on the enabled PodNet: ',
+        
+        # For disabled PodNet
+        3051: f'Failed to connect to the disabled PodNet for payload write_config: ',
+        3052: f'Failed to run write_config payload on the disabled PodNet: ',
+        3053: f'Failed to connect to the disabled PodNet for payload terminate_vpn: ',
+        3054: f'Failed to run terminate_vpn payload on the disabled PodNet: ',
+        3055: f'Failed to connect to the disabled PodNet for payload load_vpn: ',
+        3056: f'Failed to run load_vpn payload on the disabled PodNet: ',
+        3057: f'Failed to connect to the disabled PodNet for payload initiate_vpn: ',
+        3058: f'Failed to run initiate_vpn payload on the disabled PodNet: ',
+    }
+
+    # Default config_file if it is None
+    if config_file is None:
+        config_file = '/opt/robot/config.json'
+
+    # Load pod configuration
+    status, config_data, msg = load_pod_config(config_file)
+    if not status:
+        if config_data['raw'] is None:
+            return False, msg
+        else:
+            return False, msg + "\nJSON dump of raw configuration:\n" + json.dumps(config_data['raw'],
+                                                                               indent=2,
+                                                                               sort_keys=True)
+    
+    # Use the processed configuration
+    enabled = config_data['processed']['enabled']
+    disabled = config_data['processed']['disabled']
+    # Get podnet_cpe from config.json
+    podnet_cpe = config_data['processed'].get('podnet_cpe', '')
+    
+    # Get SSH username from config (default to 'robot' if not specified)
+    username = config_data.get('ssh', {}).get('username', 'robot')
+
+    # Generate VPN configuration
+    aggressive = "yes" if ike_version == "1" else "no"
+    rekey_time = int(0.9 * float(ipsec_lifetime))
+    start_action = "trap" if ipsec_establish_time == "on-traffic" else "start"
+    
+    # Build the template
```

Comment:
See the [hyperv primitive](https://github.com/CloudCIX/primitives/blob/7b0edc359e8f4c7cea8679e3cac54117f62f1c81/cloudcix_primitives/hyperv.py#L201) as an example on how to manage jinja templates.

Issue #117 has the outline of the template which should be added to the cloudcix_primitives/templates/vpns2s_ns/ path.